docs/development/style-guide.rst

```diff
@@ -0,0 +1,237 @@
+  a performance benchmark.
+* You do not need to include a changelog entry.
+
+In all other materials, avoid use of contractions only when tense can be
+confused, such as in the case of “she is gone” versus “she has gone,” etc.
+
+Hyphenation
+===========
+
+Phrasal adjectives/compound modifiers placed before a noun should be hyphenated
+to avoid confusion.
+
+Examples:
+---------
+* Astronomy-related software packages.
+* Astropy provides sustainable, high-level education to the astronomy community.
+
+Hyphenated compound words should contain hyphens in plain text, but no hyphens
+in code.
+
+Example:
+--------
+* Since most of the functionality of ``astropy`` resides in sub-packages, it is
+  best to import the desired sub-package.
+
+Numbers
+=======
+
+For numbers followed by a unit or as part of a name, use the numeral.
+
+Examples:
+---------
+* 1 arcminute
+* 32 degrees
+* Gaia data release 2 catalog
+
+For all other whole numbers, follow Associated Press (AP) style: spell out
+numbers one through nine, and use numerals for 10 and higher, with numeral-word
+combinations for millions, billions, and trillions.
+
+Examples:
+---------
+* There are two ways to build Astropy documentation.
+* Follow these 11 steps.
+* Measuring astrometry for about 2 billion stars.
+
+For casual expressions, spell out the number.
+
+Example:
+--------
+* A picture is worth a thousand words.
+
+Punctuation
+===========
+
+For consistency across Astropy materials, non-U.S. punctuation will be edited
+to reflect American punctuation preferences.
+
+**Parentheses**: punctuation belonging to parenthetical material will be placed
+inside of closing parentheses, with the exception of commas to denote a small
+pause coming after parenthetical material, and periods when parenthetical
+material is included within another sentence.
+
+Examples:
+---------
+* (For full contributor guidelines, see our documentation.)
+* Once you open a pull request (which should be opened against the ``master``
+  branch), please make sure to include the following.
+* In some cases, most of the required functionality is contained in a single
+  class (or a few classes).
+
+**Quotation marks**: periods and commas will be placed inside of closing
+quotation marks, whether double or single.
+
+Examples:
+---------
+* Chief among these terms is the concept of a “coordinate system.”
+* Because of the likelihood of confusion between these meanings of “coordinate
+  system,” `~astropy.coordinates` avoids this term wherever possible.
+
+**Hyphens vs. En Dashes vs. Em Dashes**
+
+Hyphens (-) should be used for phrasal adjectives and compound words (see
+Hyphenation above).
+
+En dashes (–) should be used for number ranges (dates, times, pages) or to
+replace the words “to” or “through,” without spaces around the dash.
+Examples:
+* See chapters 14–18
+* We’ve blocked off March 2019–May 2019 to develop a new version.
+
+Em dashes (—) can be used in place of commas, parentheses, or colons to set off
```

Comment:
The keyboard only comes with hyphen, so do I need to look up the other variants in the symbol library or something? Any tips?

Reply:
Two options: 
1. define a "compose" key and do Compose --- (good use of those pointless windows keys on your keyboard!);
2. In emacs, you can enter a special input mode with `C-\`. If you define `(setq default-input-method "TeX")` in your .emacs, that means it start to auto-interpret things like -- or \alpha, etc.

Reply:
They look the same though in the github diff, so I doubt this will be a rule that we can keep.

Reply:
There are keyboard shortcuts on a Mac, but I didn't include them because I only know the Mac shortcuts and I thought that might be too Mac-centric :) But those shortcuts are option+hyphen for an en dash, and option+shift+hyphen for the longer em dash.

I didn't realize that the different dashes don't actually render on GitHub, but would it still be useful to keep the guideline in here for reference and to help keep people in the habit of distinguishing between the different dashes?